src/controller/gateway.controller.ts

```diff
@@ -0,0 +1,54 @@
+import { FastifyInstance } from 'fastify';
+import Stripe from 'stripe';
+import { processUploadedFile } from '../services/orders.service';
+import { BadRequestError, CustomError, InternalServerError } from '../custom-errors';
+import fastifyJwt from '@fastify/jwt';
+import { AppConfig } from '../config';
+
+const XLSX_MIMETYPE = 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet';
+
+export default function (stripe: Stripe, config: AppConfig) {
+  return async function (fastify: FastifyInstance) {
+    const publicKey = Buffer.from(config.DRIVE_GATEWAY_PUBLIC_SECRET, 'base64').toString('utf8');
+
+    fastify.register(fastifyJwt, {
+      secret: {
+        public: publicKey,
+      },
+      verify: {
+        algorithms: ['RS256'],
+      },
+    });
+
+    fastify.addHook('onRequest', async (request, reply) => {
+      try {
+        await request.jwtVerify();
+      } catch (err) {
+        reply.status(401).send({ error: 'Unauthorized: Invalid token' });
+      }
+    });
+    fastify.post('/check-order-id', async (req, res) => {
+      try {
+        const file = await req.file();
```

Comment:
You need to control the size, otherwise a big file could take down the replica of the server if it refills the entire RAM

Reply:
Just check the size in number and enforce < 10MB